middle_end/flambda2/to_cmm/to_cmm_shared.ml

```diff
@@ -389,12 +409,16 @@ let make_update env res dbg ({ kind; stride } : Update_kind.t) ~symbol var
         match kind with
         | Pointer -> Word_val
         | Immediate -> Word_int
+        | Naked_int8 ->
+          if stride = Arch.size_addr then Word_int else Byte_signed
+        | Naked_int16 ->
+          if stride = Arch.size_addr then Word_int else Sixteen_signed
         | Naked_int32 ->
           (* Cmm expressions representing int32 values are always sign extended.
              By using [Word_int] in the "fields" cases (see [Update_kind],
              above) we maintain the convention that 32-bit integers in 64-bit
              fields are sign extended. *)
-          if stride > 4 then Word_int else Thirtytwo_signed
+          if stride = Arch.size_addr then Word_int else Thirtytwo_signed
```

Comment:
this change does not look like it is related to the addition of `int8` and `int16`, could you explain why it's there ?